src/libs/ModifiedExpenseMessage.ts

```diff
@@ -0,0 +1,229 @@
+import {format} from 'date-fns';
+import CONST from '@src/CONST';
+import {ReportAction} from '@src/types/onyx';
+import * as CurrencyUtils from './CurrencyUtils';
+import * as Localize from './Localize';
+import * as PolicyUtils from './PolicyUtils';
+import * as ReportUtils from './ReportUtils';
+
+/**
+ * Builds the partial message fragment for a modified field on the expense.
+ *
+ * @param newValue
+ * @param oldValue
+ * @param valueName
+ * @param valueInQuotes
+ * @param shouldConvertToLowercase
+ */
+
+function buildMessageFragmentForValue(
+    newValue: string,
+    oldValue: string,
+    valueName: string,
+    valueInQuotes: boolean,
+    setFragments: string[],
+    removalFragments: string[],
+    changeFragments: string[],
+    shouldConvertToLowercase = true,
+) {
+    const newValueToDisplay = valueInQuotes ? `"${newValue}"` : newValue;
+    const oldValueToDisplay = valueInQuotes ? `"${oldValue}"` : oldValue;
+    const displayValueName = shouldConvertToLowercase ? valueName.toLowerCase() : valueName;
+
+    if (!oldValue) {
+        const fragment = Localize.translateLocal('iou.setTheRequest', {valueName: displayValueName, newValueToDisplay});
+        setFragments.push(fragment);
+    } else if (!newValue) {
+        const fragment = Localize.translateLocal('iou.removedTheRequest', {valueName: displayValueName, oldValueToDisplay});
+        removalFragments.push(fragment);
+    } else {
+        const fragment = Localize.translateLocal('iou.updatedTheRequest', {valueName: displayValueName, newValueToDisplay, oldValueToDisplay});
+        changeFragments.push(fragment);
+    }
+}
+
+/**
+ * Get the message line for a modified expense.
+ *
+ * @param newValue
+ * @param oldValue
+ * @param valueName
+ * @param valueInQuotes
+ */
+
+function getMessageLine(prefix: string, messageFragments: string[]) {
+    if (messageFragments.length === 0) {
+        return '';
+    }
+    return messageFragments.reduce((acc, value, index) => {
+        if (index === messageFragments.length - 1) {
+            if (messageFragments.length === 1) {
+                return `${acc} ${value}.`;
+            }
+            if (messageFragments.length === 2) {
+                return `${acc} ${Localize.translateLocal('common.and')} ${value}.`;
+            }
+            return `${acc}, ${Localize.translateLocal('common.and')} ${value}.`;
+        }
+        if (index === 0) {
+            return `${acc} ${value}`;
+        }
+        return `${acc}, ${value}`;
+    }, prefix);
+}
+
+/**
+ * Get the message for a modified distance request.
+ *
+ * @param newDistance
+ * @param oldDistance
+ * @param newAmount
+ * @param oldAmount
+ */
+
+function getForDistanceRequest(newDistance: string, oldDistance: string, newAmount: string, oldAmount: string) {
```

Comment:
```suggestion
function getForDistanceRequest(newDistance: string, oldDistance: string, newAmount: string, oldAmount: string) {
```

This is all redundant and doesn't really add any value